examples/purchase.rs

```diff
@@ -0,0 +1,145 @@
+// This example demonstrates how to deal with messages and callback queries
+// within a single dialogue.
+//
+// # Example
+// ```
+// - /start
+// - Let's start! What's your full name?
+// - John Doe
+// - Select a product:
+//   [Apple, Banana, Orange, Potato]
+// - <A user selects "Banana">
+// - John Doe, product 'Banana' has been purchased successfully!
+// ```
+
+use teloxide::{
+    dispatching::dialogue::{self, GetChatId, InMemStorage},
+    prelude::*,
+    types::{InlineKeyboardButton, InlineKeyboardMarkup},
+    utils::command::BotCommands,
+};
+
+type MyDialogue = Dialogue<State, InMemStorage<State>>;
+type HandlerResult = Result<(), Box<dyn std::error::Error + Send + Sync>>;
+
+#[derive(Clone)]
+pub enum State {
+    Start,
+    ReceiveFullName,
+    ProductChosen { full_name: String },
```

Comment:
I don't think that's a good name for that state. This is the state after the user filled their name, not after the choice was made. Maybe `WaitingForProductChoice` or `ProducChoice` would be better?

Reply:
Maybe `ReceiveProductChoice` (similar to `ReceiveFullName`)?

Reply:
That works too, yes.